wordpress-mobile/WordPress-iOS
# Plans in Site Creation: Use domain_name returned from Plan Selection view

Fixes #21787

When users select a paid domain, they can still opt-in for a free wordpress.com subdomain in the plan selection view.

When a domain is selected, it is returned through `domain_name` parameter.

Use this domain when creating siteURLString instead of the one selected in the domain selection view.

## To test:

**Before testing:** Enable Plans in Site Creation feature flag

4 cases need to be tested and confirmed to be working properly:
- Select a paid domain, then select a free plan. Confirm that the domain shown in the pop-up matches the one that the site was created with
- Select a paid domain, then select a paid plan. Confirm that checkout is opened and that the domain selected in the domain selection matches the one that appears in the checkout view
- Select a free domain, then select a free plan. Confirm that the site was created with the domain that was selected in the domain selection view
- Select a free domain, then select a paid plan. Confirm that the site was created with the domain that was selected in the domain selection view

## Regression Notes
1. Potential unintended areas of impact

The change in `siteURLString` could potentially break existing functionality. Need to retest all the scenarios to make sure everything works as expected.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Updated unit tests

3. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:


UI Changes testing checklist:
